pkg/api/validation/validation.go

```diff
@@ -1787,6 +1787,49 @@ func validateProbe(probe *api.Probe, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateOrSetServiceAffinityConfig(affinity api.ServiceAffinity, config *api.ServiceAffinityConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// No need to validate when no session affinity.
+	if affinity == api.ServiceAffinityNone {
+		return allErrs
+	}
+	if affinity == api.ServiceAffinityClientIP {
+		// If ServiceAffinityConfig is nil, set the default value
+		if config == nil {
```

Comment:
Defaulting happens in API defaulting code, not validation.

./pkg/api/v1/defaults.go